pkg/appliance/functions.go

```diff
@@ -776,3 +788,251 @@ func PrettyBytes(v float64) string {
 	}
 	return fmt.Sprintf("%.2fYB", v)
 }
+
+type ManifestConfig struct {
+	MediaType string
+	Size      int
+	Digest    string
+}
+type DockerManifest struct {
+	SchemaVersion int
+	MediaType     string
+	Config        ManifestConfig
+	Layers        []ManifestConfig
+}
+
+type ImageJSON struct {
+	Image string `json:"image"`
+}
+
+func DownloadDockerBundles(ctx context.Context, out io.Writer, client *http.Client, zipName, registry string, images map[string]string, ciMode bool) (*os.File, fs.FileInfo, error) {
+	// Create zip-archive
+	archive, err := os.Create(fmt.Sprintf("/tmp/%s.zip", zipName))
+	if err != nil {
+		return nil, nil, err
+	}
+	defer archive.Close()
+	zipWriter := zip.NewWriter(archive)
+	defer zipWriter.Close()
+	fileStat, err := archive.Stat()
+	if err != nil {
+		return nil, nil, err
+	}
+
+	log.Info("downloading image layers for ", zipName)
+	var p *tui.Progress
+	if !ciMode {
+		p = tui.New(ctx, out)
+		defer p.Wait()
+	}
+	errChan := make(chan error)
+	var wg sync.WaitGroup
```

Comment:
https://pkg.go.dev/golang.org/x/sync/errgroup could work here as a option, that way we get rid of errChan, since we don't want to continue if any of these downloads gets  error. that would reduce complexity and func signature